services/QuillLMS/app/queries/teacher_dashboard_metrics.rb

```diff
@@ -0,0 +1,33 @@
+class TeacherDashboardMetrics
+
+  def self.queries(user)
+    today = Date.today
+    days_since_last_sunday = today.wday == 0 ? 0 : (today.wday + 6) % 7 + 1
+    last_sunday = today - days_since_last_sunday
+    july_first_of_this_year = Date.parse("01-07-#{today.year}")
+    last_july_first = today.month > 7 ? july_first_of_this_year : july_first_of_this_year - 1.year
+    classroom_ids = user.classrooms_i_teach.map(&:id)
+    {
+      weekly_assigned_activities_count: count_of_assigned_activities(classroom_ids, last_sunday),
+      yearly_assigned_activities_count: count_of_assigned_activities(classroom_ids, last_july_first),
+      weekly_completed_activities_count: count_of_completed_activities(classroom_ids, last_sunday),
+      yearly_completed_activities_count: count_of_completed_activities(classroom_ids, last_july_first)
+    }
+  end
+
+  def self.count_of_assigned_activities(classroom_ids, start_date)
+    classroom_units = ClassroomUnit.where(classroom_id: classroom_ids).where("created_at >= ?", start_date)
+    total_count = 0
```

Comment:
You could get rid of the temporary variable `total_count` with the enumerable method `reduce`:

```
classroom_units.reduce(0) do |total_count, classroom_unit|
  number_of_students = classroom_unit.assigned_student_ids.length
  number_of_activities = UnitActivity.where(unit_id: classroom_unit.unit_id).count
  total_count += number_of_students * number_of_activities
end
```